src/compile/unit.ts

```diff
@@ -83,11 +83,18 @@ export class UnitModel extends Model {
 
   private _initScale(mark: Mark, encoding: Encoding, config: Config): Dict<Scale> {
     return UNIT_SCALE_CHANNELS.reduce(function(_scale, channel) {
-      if (vlEncoding.has(encoding, channel)) {
-        const scaleSpec = encoding[channel].scale || {};
-        const channelDef = encoding[channel];
+      if (vlEncoding.has(encoding, channel) ||
+          (channel === X && vlEncoding.has(encoding, X2)) ||
+          (channel === Y && vlEncoding.has(encoding, Y2))) {
+
+        let targetChannel = channel;
+        if (!vlEncoding.has(encoding, channel)) {
+          targetChannel = channel === X ? X2 : Y2;
+        }
+        const channelDef = encoding[targetChannel];
+        const scaleSpec = channelDef.scale || {};
 
-        const _scaleType = scaleType(scaleSpec, channelDef, channel, mark);
+        const _scaleType = scaleType(scaleSpec, channelDef, targetChannel, mark);
```

Comment:
> If you read the logic inside scaleType, I don't think this should be targetChannel here.

Ah you are right.

> Maybe we have to generate warning if they do not.
Yeah I think that would be a good idea. Although, it would just ignore the `x2`'s `type`, `bin`, `timeUnit`. That should make the spec work, but yeah a warning would be nice